x-pack/solutions/observability/plugins/observability_logs_explorer/public/redirect_to_discover.tsx

```diff
@@ -25,9 +32,38 @@ export const renderDiscoverRedirect = (core: CoreStart, appParams: AppMountParam
   };
 };
 
-export const DiscoverRedirect = ({ core }: { core: CoreStart }) => {
+export const DiscoverRedirect = ({
+  core,
+  discover,
+}: {
+  core: CoreStart;
+  discover: DiscoverStart;
+}) => {
   const location = useLocation();
-  const path = `${location.pathname}${location.search}`;
-  core.application.navigateToApp(DISCOVER_APP_ID, { replace: true, path });
+  const searchParams = new URLSearchParams(location.search);
+  const pageStateEncoded = searchParams.get('pageState') || '';
+  const rawPageState = safeDecode(pageStateEncoded) || {};
+
+  const parsedState = normalizeUrlState(rawPageState);
+
+  if (parsedState) {
+    const discoverParams: DiscoverAppLocatorParams = {
+      timeRange: parsedState.time,
+      refreshInterval: parsedState.refreshInterval,
+      filters: parsedState.filters,
+      query: parsedState.query,
+      breakdownField: parsedState.chart?.breakdownField ?? undefined,
+      columns: parsedState.grid?.columns?.map((column) => column.field),
```

Comment:
Does this account for the different column types? Not all have a single `field` if I remember correctly.